src/mlpack/core/dists/discrete_distribution.hpp

```diff
@@ -167,7 +168,30 @@ class DiscreteDistribution
     // TODO: consider storing log probabilities instead?
     return log(Probability(observation));
   }
+  
+  /**
+   * Calculates the discrete probability density function for each
+   * data point (column) in the given matrix.
+   *
+   * @param x List of observations.
+   * @param probabilities Output probabilities for each input observation.
+   */
+  void Probability(const arma::mat& x, arma::vec& probabilities) const
+  {
+    arma::vec logProbabilities;
+    LogProbability(x, logProbabilities);
```

Comment:
It would be better to just call `Probability()` individually with each point here, to avoid converting to log-space and then back each time. (It would also be slightly faster.)